GitUI/Avatars/FileSystemAvatarCache.cs

```diff
@@ -152,10 +151,15 @@ public async Task ClearCacheAsync()
                                 // do nothing
                             }
                         }
-                    });
+                    }).ContinueWith(_ =>
+                    {
+                        CacheCleared?.Invoke(this, EventArgs.Empty);
+                    }, TaskScheduler.Current);
```

Comment:
I cannot see anything actually async in this entire function `ClearCacheAsync`. Rather remove the poinless `await Task.Run`.
(Please use `await ...` in order to switch between background and UI threads as a basic principle.)

Reply:
It is inherited from interface (and was not introduced in this PR).
I just built on existing code but I'm OK to remove async everywhere else...

Reply:
> I imagine it was run in a task to not block the UI while deleting all the files...

The old code
`ThreadHelper.JoinableTaskFactory.Run(AvatarService.CacheCleaner.ClearCacheAsync);`
did not switch to a background thread IIUIC.
`ThreadHelper.FileAndForget` should rather be used there.
And `await Task.Run` be removed here. 

At least on a background thread, `.ContinueWith(..., TaskScheduler.Current)` is pointless, isn't it?